Orbitersdk/samples/ProjectApollo/src_sys/toggleswitch.cpp

```diff
@@ -417,6 +417,14 @@ bool TwoPositionSwitch::CheckMouseClickVC(int event, VECTOR3 &p)
 	return DoCheckMouseClickVC(event, p);
 }
 
+void TwoPositionSwitch::VesimSwitchTo(int newState)
```

Comment:
I think you can achieve the same result by using the existing TwoPositionSwitch::SetState function here.

Reply:
The idea was to have a general method for manipulating a switch, similar to CheckMouseClickVC and CheckMouseClick, but without the requirement to generate mouse click event or other unnecessary stuff. So what I wanted is a common function which processes a "deus ex machina" switching. Maybe my choice for the name wasn't the best, because such a method could be relevant outside of the Vesim bindings. Even the implementations of CheckMouseClickVC and CheckMouseClick would cleaner if they wouldn't be so redundant, and rather after the processing of the click events they would call the same method for the actual logic of the switch. Certainly it would be neat to extend this for all the buttons and switches, but I wanted to minimize the changes of the toggleswitch.cpp on first approximation.
On this specific TwoPositionSwitch::VesimSwitchTo(int newState): the TwoPositionSwitch::SetState does not check the attempt to set the switch to the same state as in it was originally. So it should be checked before each call of this function (there are four of them at the moment). It would result in redundant code, but additionally it seems to me that this type of logic is rather the part of the switch behaviour and as such, should be handled there.

Reply:
SwitchTo does check if you're trying to set the same state. If you are, it exits out and SetState won't play the sound. No issue there.
I agree with you that its nice to have this as a generic function. If you want, you can give them a nice generic name. In either case I''ll approve it.